COMET/misc_plugins/PlotScripts/analysis_scripts/IV_PQC.py

```diff
@@ -1,82 +1,670 @@
+        self.PlotDict["BasePlots_diode"] += self.basePlots_2
+
+        # Add full depletion point to 1/c^2 curve
+        if self.config["IV_PQC"].get("1C2", {}).get("DoFullDepletionCalculation", False):
+            try:
+                if self.basePlots4.Overlay.A_1C2.children:
+                    c2plot = self.basePlots4.Overlay.A_1C2.opts(clone=True)
+                else:
+                    c2plot = self.basePlots4.Curve.A_1C2.opts(clone=True)
+                fdestimation = self.find_full_depletion_c2(c2plot, self.data, self.config, diode_files, PlotLabel="Full depletion estimation")
+            except Exception as err:
+                self.log.warning("No full depletion calculation possible... Error: {}".format(err))
+
+        # Find resistivity
+        C_min = np.mean(self.data[df]['data']['Capacity'][-20:])
+        d_active = self.config['IV_PQC_parameter']['epsilonNull'] * self.config['IV_PQC_parameter']['epsilonSiliconOxide'] * (float(self.data[df]['header'][0].split(':')[1]) * (1e-8)) * (1e-2) / C_min  # in cm
+        T_n = 295 / 300
+        u_holes_mobility = 54.3 * pow(T_n, -0.57) + 1.36 * (1e+8) * pow(295, -2.23) / (1 + ((5e+12) / (2.35 * (1e+17) * pow(T_n, 2.4))) * 0.88 * pow(T_n, -0.146))  # in cm^2/(V*s)
+        rho = d_active * d_active / (2 * self.config['IV_PQC_parameter']['epsilonNull'] * (1e-2) * self.config['IV_PQC_parameter']['epsilonSiliconOxide'] * fdestimation[1] * u_holes_mobility)  # in Ohm * cm
+        rho_table = '{:.2e}'.format(rho)  # value to show later on in the table showing the results of the analysis
+        resistivity.append(rho_table)
+
+        # Add a table that show the results of the analysis
+        if len(diode_files) == 1:
+            self.PlotDict["BasePlots_diode"] += fdestimation[0]
+            # Add trial plots
+            self.PlotDict["BasePlots_diode"] += capacity_curve
+            ##self.PlotDict["BasePlots_diode"] += derivative_onec2_curve * deronec2_savgol_plot
+            ##self.PlotDict["BasePlots_diode"] += deronec2_savgol_plot
+            # Add table
+            df3 = pd.DataFrame({"Name": diode_files, "full depletion voltage (V)": fdepvoltage, " Bulk resistivity (Ohm * cm)": resistivity})
+            table2 = hv.Table(df3, label='Diode analysis')
+            table2.opts(width=1300, height=800)
+            self.PlotDict["BasePlots_diode"] += table2
+
+        else:
+            df3 = pd.DataFrame({"Name": diode_files, "full depletion voltage (V)": fdepvoltage, "Bulk resistivity (Ohm * cm)": resistivity})
+            table2 = hv.Table(df3, label='Diode analysis')
+            table2.opts(width=1300, height=800)
+            self.PlotDict["BasePlots_diode"] += table2
+
+        return self.PlotDict["BasePlots_diode"]
+
+    def analysis_gate(self, df):
```

Comment:
![Codacy](https://app.codacy.com/assets/images/favicon.png) Issue found: [Global variable 'maxsavgol' undefined at the module level](https://app.codacy.com/manual/dominic.bloech/UniDAQ/pullRequest?prid=5481807)